overlord/devicestate/devicestate.go

```diff
@@ -188,7 +189,11 @@ func checkGadgetOrKernel(st *state.State, snapInfo, curInfo *snap.Info, flags sn
 	return nil
 }
 
-func init() {
-	snapstate.AddCheckSnapCallback(checkGadgetOrKernel)
+var once sync.Once
+
+func delayedCrossMgrInit() {
+	once.Do(func() {
+		snapstate.AddCheckSnapCallback(checkGadgetOrKernel)
+	})
 	snapstate.CanAutoRefresh = canAutoRefresh
```

Comment:
any reason not to include the assignment in the once?

Reply:
the assignments are idempotent, same value all the time